pyproject.toml

```
@@ -9,5 +9,5 @@ requires = [
   "numpy>=2.0.0;  python_version>='3.9'",
   "pip",
   "scikit-build>=0.14.0",
-  "setuptools==59.2.0",
+  "setuptools>=59.2.0",
```

I normally wouldn't recommend pinning setuptools on an upperbound unless you have an explicit reason, but v71 changed how it vendors its dependencies in a major way. Where it'll prefer already installed deps over its vendors. This can cause issues if the environment has an older version of a dep installed (which wasn't problematic before for setuptools, as it'd use it's own vendor anyway).

Given the various distributions and dockerization of opencv-python, it might be warranted to progressively relax this pin by first going to:
```suggestion
  "setuptools>=59.2.0,<71.0.0",
```

Leaving this to the maintainers whether that's a concern 😄

---

I'm reticent to introduce an upper-bound with neither any observed problems nor a plan to fix any anticipated problems. But I'm happy to commit the suggestion if maintainers agree it's prudent.

---

Yeah same, figured I'd at least let it be known.

---

I just tried on macOS in a virtual environment and it fails:

```
Collecting setuptools<71.0.0,>59.2.0 (from -r requirements.txt (line 2))
  Using cached setuptools-70.3.0-py3-none-any.whl.metadata (5.8 kB)
Collecting opencv_python>=4.5.2.52 (from -r requirements.txt (line 4))
  Using cached opencv-python-4.10.0.84.tar.gz (95.1 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
...
  File "/private/var/folders/vp/8rjwz51s5wx5dptcn89czk8m0000gn/T/pip-build-env-h9pxukyj/overlay/lib/python3.12/site-packages/setuptools/__init__.py", line 10, in <module>
    import distutils.core
ModuleNotFoundError: No module named 'distutils'
```

I'm on macOS 11.7.10, Python 3.12.5. I am upgrading to Python 3.12.6 now to see if it makes a difference.

---

@mm3509 Since that doesn't pertain to @Avasam's concern about setuptools v71 and caching, it doesn't belong in this thread, and please move the discussion to the main comments section below.

---

@bryankaplan Done, thank you